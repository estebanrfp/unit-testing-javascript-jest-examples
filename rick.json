{
    "id": 1,
    "name": "Rick Sanchez",
    "status": "human",
    "species": "species",
    "gender": "gender"
}