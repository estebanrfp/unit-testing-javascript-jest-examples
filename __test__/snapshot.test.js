import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });
    test('Siempre fallará la instantanea', () => {
        const user = {
            createAt: new Date,
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot();
    });

    test('Tenemos una excepsión dentro del código', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Esteban Fuster Pozzi"
        }
        expect(user).toMatchSnapshot({ id: expect.any(Number) });
    });
});

// jest -u (update snapshots)
