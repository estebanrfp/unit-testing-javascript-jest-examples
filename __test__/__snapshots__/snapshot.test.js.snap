// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "gender",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "species",
  "status": "human",
}
`;
